### DISCUSSION TITLE
Add custom CSS conditionally

### DISCUSSION THREAD

**Discussion by halwesit** [directus/directus#21694] [Feature Requests]
### Summary

Hello everyone,
I have a special request, which is I think it may helpful for others as well if we have a custom CSS conditionally. 

### Basic Example

For example I want to change the primary color based on different user roles, adding different fonts for different languages.

### Motivation

I want to add RTL support some users based on thier roles, or thier language preference with my custom code that made Directus us support RTL. 

### Detailed Design

... 

### Must Have:
- A
- B

### Should Have:
- C

### Could Have:
- D

### Won't Have:
- E

### Drawbacks

... 

### Alternatives

.... 

### Adoption Strategy

...

**Comment by rijkvanzanten** [comment]
I think my current preferred way to achieve that is through theming, and then allow you to set the theme through a role. Custom CSS is always going to be a little brittle as the DOM isn't guaranteed to stay the same release to release

**Reply by halwesit** [reply]
By the way for custom theming, we should go deeper for make themes rtl support, because by current options we couldn't able to do that, but by custom css we can.

**Reply by rijkvanzanten** [reply]
RTL is something that really needs to be baked in from the start in the base components. Even with custom CSS the amount of stuff that would have to be overwritten to do a proper RTL theme will be very large and hard to maintain. I'd like to natively support it in the future, but it has been really hard to include as it's such a small percentage of the user base who would be benefitted by it :C (... which is a bit of a catch-22: without RTL support, RTL users will go elsewhere 🔁)

**Reply by halwesit** [reply]
Let me aware you of the importance of RTL languages. because if it's something that needs to be baked in from the start in the base components, it is worth being added.

1. Adding RTL support makes the project accessible and usable for people who use RTL languages like [Arabic, Kurdish, Hebrew, Persian, and Urdu, which is used by more than 748 million people](https://lingohub.com/blog/right-to-left-vs-left-to-right). This is a significant user base that cannot be overlooked. 
2. For RTL language users, interfaces without RTL support can be challenging to use and understand. Proper RTL support significantly improves the user experience for this audience.
3. By supporting RTL languages, you can reach and expand into new markets that predominantly use these languages.
4. Many similar projects still lack proper RTL support. Offering this feature could give the project a competitive advantage.
5. Localization (often referred to as “l10n”) is the process of adapting a product or content to a specific locale or market. Adding RTL support is a crucial part of this process for RTL languages.

I enjoy working and using Directus for my projects, but it needs to support RTL scripts to let us choose Directus for all projects without worrying about user experience.

**Reply by rijkvanzanten** [reply]
I'm very aware of all that points, but that doesn't necessarily negate my original concern around the cost of implementation..

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
